iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -247,19 +249,19 @@ TEST_F(ServiceDiscovery_test, OfferDifferentServicesWithSameInstanceAndEvent)
     ASSERT_THAT(serviceContainer.size(), Eq(1U));
     EXPECT_THAT(*serviceContainer.begin(), Eq(SERVICE_DESCRIPTION1));
     sut.findService(SERVICE_DESCRIPTION1.getServiceIDString(), INSTANCE, EVENT, findHandler);
-    compareServiceContainers(serviceContainer, searchResultOfFindServiceWithFindHandler);
+    compareAndClearServiceContainers(serviceContainer, searchResultOfFindServiceWithFindHandler);
 
     serviceContainer = sut.findService(SERVICE_DESCRIPTION2.getServiceIDString(), INSTANCE, EVENT);
     ASSERT_THAT(serviceContainer.size(), Eq(1U));
     EXPECT_THAT(*serviceContainer.begin(), Eq(SERVICE_DESCRIPTION2));
     sut.findService(SERVICE_DESCRIPTION2.getServiceIDString(), INSTANCE, EVENT, findHandler);
-    compareServiceContainers(serviceContainer, searchResultOfFindServiceWithFindHandler);
+    compareAndClearServiceContainers(serviceContainer, searchResultOfFindServiceWithFindHandler);
```

Comment:
Multiple calls is why we clear the containers before, right? Usually I would avoid combining comparisons with modifying operations like this, but it is not called often and avoids some explicit clear calls.

Reply:
Correct. Usually I would agree but for such simple tests I wanted to avoid calling compare and then clear in almost every test.

Reply:
I'll refactor `compareAndClearServiceContainers` in #1065